### DISCUSSION TITLE
Could component blueprint `Thot` be made optional?

### DISCUSSION THREAD

**Discussion by keckler** [terrapower/armi#1432] [Ideas]
The fact that the `Thot` attribute of component blueprints is _mandatory_ has always seemed a bit odd to me.

Wouldn't it be possible to just resolve `Thot` to `Tinput` if the user doesn't enter a `Thot` value? `Tinput` is obviously definitely necessary. But there seem to be at least a couple common scenarios when `Thot` isn't really needed:
(1) when a user is running thermal-hydraulics calculations as part of their simulation, so the component temperatures will be updated anyways
(2) when a model is room temperature, so `Thot == Tinput` for every single component

Adding in some logic to resolve `Thot` to `Tinput` when the user doesn't explicitly input `Thot` would appear to be easy to do, and it could help cut down on the size of blueprints files by a decent chunk.

**Comment by john-science** [comment]
Would it be hard to make it clear to people that `Thot` is assumed if not entered explicitly?

**Reply by keckler** [reply]
As far as I can tell, the only place that we ever tell the user what `Thot` means is here: https://terrapower.github.io/armi/user/inputs/blueprints.html#defining-a-component

We could easily add a sentence to that description about the assumption.

**Comment by keckler** [comment]
At to the list of places where `Thot` is not needed:
(3) When the material is `Void`, which has no material properties or isotopics
